lib/matplotlib/cbook/__init__.py

```diff
@@ -367,6 +367,18 @@ def process(self, s, *args, **kwargs):
                     self._remove_proxy(proxy)
 
 
+class EventEmitter(object):
```

Comment:
Agree, when I get back I can create a separate PR that adds this functionality in and adds it across the board.